File: TransCoder_tokenized_test_set_functions/python/NUMBER_SUBARRAYS_SUM_EXACTLY_EQUAL_K.py
def findSubarraySum ( arr, n, Sum ) :
    prevSum = defaultdict ( lambda : 0 )
    res = 0
    currsum = 0
    for i in range ( 0, n ) :
        currsum += arr [ i ]
        if currsum == Sum :
            res += 1
        if ( currsum - Sum ) in prevSum :
            res += prevSum [ currsum - Sum ]
        prevSum [ currsum ] += 1
    return res

File: TransCoder_tokenized_test_set_functions/java/NUMBER_SUBARRAYS_SUM_EXACTLY_EQUAL_K.java
static int findSubarraySum ( int arr [ ], int n, int sum ) {
  HashMap < Integer, Integer > prevSum = new HashMap < > ( );
  int res = 0;
  int currsum = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    currsum += arr [ i ];
    if ( currsum == sum ) res ++;
    if ( prevSum . containsKey ( currsum - sum ) ) res += prevSum . get ( currsum - sum );
    Integer count = prevSum . get ( currsum );
    if ( count == null ) prevSum . put ( currsum, 1 );
    else prevSum . put ( currsum, count + 1 );
  }
  return res;
}